x-pack/plugins/observability/public/pages/alert_details/components/alert_details.tsx

```diff
@@ -27,19 +31,31 @@ import { paths } from '../../../config/paths';
 
 export function AlertDetails() {
   const {
+    uiSettings,
     http,
     cases: {
       helpers: { canUseCases },
       ui: { getCasesContext },
     },
+    triggersActionsUi: { ruleTypeRegistry },
   } = useKibana<ObservabilityAppServices>().services;
+
   const { ObservabilityPageTemplate, config } = usePluginContext();
   const { alertId } = useParams<AlertDetailsPathParams>();
   const [isLoading, alert] = useFetchAlertDetail(alertId);
-
+  const [ruleTypeModel, setRuleTypeModel] = useState<RuleTypeModel | null>(null);
   const CasesContext = getCasesContext();
   const userCasesPermissions = canUseCases();
+  const { rule } = useFetchRule({
+    ruleId: alert?.fields[ALERT_RULE_UUID],
```

Comment:
Shouldn't we fetch the rule when we have alert fields (and as a result the ruleId)?
Then we don't need to make ruleId parameter in useFetchRule optional.

Reply:
@maryam-saeidi, yeah there is a custom hooks dependency here.